minio_storage/storage.py

```diff
@@ -388,82 +389,134 @@ def create_minio_client_from_settings(*, minio_kwargs=None):
 
 @deconstructible
 class MinioMediaStorage(MinioStorage):
-    def __init__(self):
-        client = create_minio_client_from_settings()
-        bucket_name = get_setting("MINIO_STORAGE_MEDIA_BUCKET_NAME")
-        base_url = get_setting("MINIO_STORAGE_MEDIA_URL", None)
-        auto_create_bucket = get_setting(
-            "MINIO_STORAGE_AUTO_CREATE_MEDIA_BUCKET", False
-        )
-        auto_create_policy = get_setting(
+    def __init__(  # noqa: C901
+        self,
+        *,
```

Comment:
reordering arguments and requiring keyword arguments is an backwards incompatible change and maybe should be mentioned in the (badly maintained) changelog.

I think it should be fine. At least it breaks early if someone is using it with positional arguments now.